### PR TITLE
add : staking contracts from old repo and integritiy with its libraries

### PR DESCRIPTION
**Pull request by aliberkhsl**
# Description

Moved staking contracts from the old repo and integrate dependencies with newer versions

### CODE REVIEW

File: test/staking/SimpleDAIStaking.test.ts [hunk 1]
```diff
@@ -467,63 +524,40 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
...
+      transaction.events.find(_ => _.event === "StakeWithdraw").args.staker
+    ).to.be.equal(staker.address);
+    expect(
+      transaction.events
+        .find(_ => _.event === "StakeWithdraw")
+        .args.value.toString()
+    ).to.be.equal((stakerDaiBalanceAfter - stakerDaiBalanceBefore).toString());
   });
 
-  it("should be able to withdraw stake by staker when the worth is lower than the actual staked", async () => {
```
Comment (sirpy) [inline test/staking/SimpleDAIStaking.test.ts]: why was this test removed?

File: test/staking/SimpleDAIStaking.test.ts [hunk 2]
```diff
@@ -532,257 +566,203 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
...
-        from: staker
-      })
+    const error = await goodCompoundStaking
+      .connect(staker)
+      .stake("0", 100)
       .catch(e => e);
-    expect(error.message).to.have.string("You need to stake a positive token amount");
-  });
-
-  it("should be able to stake dai when the allowed dai amount is higher than the staked amount", async () => {
```
Comment (sirpy) [inline test/staking/SimpleDAIStaking.test.ts]: why was this removed?

File: test/staking/SimpleDAIStaking.test.ts [hunk 3]
```diff
@@ -532,257 +566,203 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
...
-    await simpleStaking.withdrawStake(weiAmount, {
-      from: staker
-    });
-  });
+    expect(transaction.events.find(_ => _.event === "Staked")).not.to.be.empty;
+    expect(
+      transaction.events.find(_ => _.event === "Staked").args.value.toString()
+    ).to.be.equal(weiAmount.toString());
 
-  it("should not withdraw interest to owner if cDAI value is lower than the staked", async () => {
```
Comment (sirpy) [inline test/staking/SimpleDAIStaking.test.ts]: why?

File: test/staking/SimpleDAIStaking.test.ts [hunk 4]
```diff
@@ -532,257 +566,203 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
...
-    const fundBalanceAfter = await cDAI.balanceOf(founder);
-    expect(cdaiGains.toString()).to.be.equal("0");
-    expect(precisionLossDai.toString()).to.be.equal("0");
-    expect(fundBalanceAfter.toString()).to.be.equal(fundBalanceBefore.toString());
-    await simpleStakingFM.withdrawStake(weiAmount, {
-      from: staker
-    });
-  });
-
-  it("should not be able to stake if the getting an error while minting new cdai", async () => {
```
Comment (sirpy) [inline test/staking/SimpleDAIStaking.test.ts]: why?

File: test/staking/SimpleDAIStaking.test.ts [hunk 5]
```diff
@@ -792,354 +772,352 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
...
+      simpleStaking1.address,
+      encodedCall,
+      avatar,
+      0
+    );
+    let balanceAfter = await cdai1.balanceOf(avatar);
     expect(balanceAfter.toString()).to.be.equal(balanceBefore.toString());
   });
 
-  it("should not be able to change the fund manager address if not owner", async () => {
```
Comment (sirpy) [inline test/staking/SimpleDAIStaking.test.ts]: can we test that if fundmanager changes in nameservice it still works as expected?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: test/staking/SimpleDAIStaking.test.ts (hunk 4 above)
Comment (aliberkhsl) [reply test/staking/SimpleDAIStaking.test.ts]: Because we cant get it fail according to our code right now with cDAIMock

File: test/staking/SimpleDAIStaking.test.ts (hunk 1 above)
Comment (sirpy) [reply test/staking/SimpleDAIStaking.test.ts]: @aliberkhsl
